### PR TITLE
Adds date selection keybindings

### PR DESCRIPTION
**Pull request by Vigasaurus**
### Changes

- Adds Keybindings for date selection, as per discussion in #625 

I will say, this probably needs some solid stress testing on staging or similar, since I've encountered quite a few edge cases where the menu has remained open, or Flatpickr hasn't opened, stuff like that. I think I've got most of them resolved, but there could be some remaining I suppose. 

### Tests
- [X] This PR does not require tests

### Changelog
- [x] Entry has been added to changelog

### Documentation
- Marko said he might add a `Keyboard Shortcuts` docs page

### Screenshots
image

### CODE REVIEW

File: assets/js/dashboard/datepicker.js [hunk 1]
```diff
@@ -285,15 +296,36 @@ class DatePicker extends React.Component {
 
     const date = opts.date ? formatISO(opts.date) : false;
 
+    let keybind = '';
+
+    switch (text) {
```
Comment (ukutaht) [inline assets/js/dashboard/datepicker.js:301]: I much prefer to define a mapping at the top:
```javascript
const KEYBINDS = {
  'Today': 'D',
  'Realtime': 'R'
}
```

etc. and then look it up here `const keybind = KEYBINDS[text]`

File: assets/js/dashboard/datepicker.js [hunk 2]
```diff
@@ -96,7 +98,16 @@ class DatePicker extends React.Component {
       }
     }
 
-    if (newSearch.date) {
+    this.setState({open: false});
+
+    const keys = ['d', 'r', 'w', 'm', 'y', 'D', 'R', 'W', 'M', 'Y'];
```
Comment (ukutaht) [inline assets/js/dashboard/datepicker.js:103]: Probably easier to just `downCase` or `upperCase` the input instead of defining keys twice.

**Review by ukutaht — CHANGES_REQUESTED** [review]

**Review by ukutaht — APPROVED** [review]
